### PR TITLE
Made Calculator

### PR DESCRIPTION
**Pull request by rahul72parihar**
Javascript
- made 3 variable for num1, operator and num2
- different function for clear and equal 
- made a display with heading and textContent to display

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+const display = document.getElementById("display");
+const equalBtn = document.getElementById("equal");
+const clearBtn = document.getElementById("clear");
+let num1 = null;
+let num2 = null;
+let op = null;
+console.log("render");
```
Comment (rahul72parihar) [inline index.js]: Remove console.log

File: index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const display = document.getElementById("display");
+const equalBtn = document.getElementById("equal");
+const clearBtn = document.getElementById("clear");
+let num1 = null;
+let num2 = null;
+let op = null;
```
Comment (rahul72parihar) [inline index.js]: give better name to op

File: styles.css [hunk 3]
```diff
@@ -0,0 +1,38 @@
+*,
```
Comment (rahul72parihar) [inline styles.css:1]: add newline after each selector

File: index.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+const display = document.getElementById("display");
+const equalBtn = document.getElementById("equal");
+const clearBtn = document.getElementById("clear");
+let num1 = null;
+let num2 = null;
+let mainOperator = null;
```
Comment (balwant-jtc) [inline index.js]: Use array destructuring to initialize the variables num1, num2, and mainOperator to null.
`let [num1, num2, mainOperator] = [null, null, null];`

File: index.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+//Handle Clicks on Operators and Numbers
+document.addEventListener("click", handleClick);
+function handleClick(e) {
+  if (e.target.classList.contains("number")) {
+    handleNumber(parseInt(e.target.value));
+  } else if (e.target.classList.contains("operator")) {
+    handleOperator(e.target.value);
+  }
+}
+function handleNumber(num) {
```
Comment (balwant-jtc) [inline index.js]: Can you simplify this function, its hard to read currently?

File: index.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      num1 += num;
+    }
+    display.textContent = num1;
+  } else {
+    if (num2 === null) num2 = num;
+    else {
+      num2 *= 10;
+      num2 += num;
+    }
+    display.textContent = num1 + mainOperator + num2;
```
Comment (balwant-jtc) [inline index.js]: Use template literals to concate multiple strings.

File: index.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+}
+//handle Click on Equal Button
+equalBtn.addEventListener("click", handleEqual);
+function handleEqual() {
+  if (num1 === null || mainOperator === null || num2 === null) return;
+  if (mainOperator === "+") num1 = num1 + num2;
+  else if (mainOperator === "-") num1 -= num2;
+  else if (mainOperator === "*") num1 *= num2;
+  else if (mainOperator === "/") num1 /= num2;
+  num2 = null;
```
Comment (balwant-jtc) [inline index.js]: `[num2, mainOperator] = [null, null];`

File: index.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  mainOperator = null;
+  display.textContent = num1;
+}
+//Handle Clear Button
+clearBtn.addEventListener("click", handleClear);
+function handleClear() {
+  num1 = null;
```
Comment (balwant-jtc) [inline index.js]: [num1, mainOperator, num2] = [null, null, null];

File: index.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  if (num2 !== null) handleEqual();
+  mainOperator = operator;
+  display.textContent = num1 + mainOperator;
+}
+//handle Click on Equal Button
+equalBtn.addEventListener("click", handleEqual);
+function handleEqual() {
+  if (num1 === null || mainOperator === null || num2 === null) return;
+  if (mainOperator === "+") num1 = num1 + num2;
+  else if (mainOperator === "-") num1 -= num2;
```
Comment (balwant-jtc) [inline index.js]: use `switch` here instead of `if` to make it more readable

**Review by balwant-jtc — CHANGES_REQUESTED** [review]

**Review by balwant-jtc — APPROVED** [review]

**Pull request merged**